### PR TITLE
Improve the ldap_ban playbook

### PR DESCRIPTION
**Pull request by KellerFuchs**
- [x] Use `ldap_attr` rather than running a raw command on `ldap.hashbang.sh`.
- [x] Ignore errors from `sss_cache`.
  It turns out it can spuriously error-out.  Wheee, quality software \o/
- [x] Support banning multiple users at once.
- [x] Support (optionally) deleting the users' home directories.
- [x] Prompt the administrator interactively for options.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Review by ghost — DISMISSED** [review]
"Missing required 'ldap' module". Seems like python-ldap needs to be installed? Yay, CoreOS!

**Comment by KellerFuchs** [comment]
@dpflug No, it needs `python-ldap` on your machine. (`apt install python-ldap` on Debian)

File: README.md [hunk 1]
```diff
@@ -61,10 +61,14 @@ There are several playbooks present here:
 - `mail.yml` deploy the mail aliases and Postfix configuration.
 - `irc.yml` deploys static and templated configuration to the IRC servers,
   including oper blocks for users defined in `group_vars/all/users.yml`.
-- `ldap_ban.yml` disables a user's account in LDAP and terminate their
+- `ldap_ban.yml` disables a user accounts in LDAP and terminate their
```
Comment (benharri) [inline README.md]: remove "a"?

File: ldap_ban.yml [hunk 2]
```diff
@@ -1,37 +1,64 @@
...
-      changetype: modify
-      replace: loginShell
-      loginShell: /usr/sbin/nologin
+- name: Parse parameters
+  hosts: ldap.hashbang.sh,shell_servers
+  gather_facts: no
 
+  vars_prompt:
+    - name: users
+      prompt: Colon-separated list of users to ban
```
Comment (benharri) [inline ldap_ban.yml]: Comma

File: ldap_ban.yml [hunk 3]
```diff
@@ -1,37 +1,64 @@
...
-    - name: Terminate sessions for {{ user }}
-      command: loginctl terminate-user {{ user }}
+    - name: Terminate user sessions
+      command: loginctl terminate-user {{ item }}
       register: terminate
+      with_items: "{{ user_list }}"
       failed_when: |
         'Could not terminate user: No user' not in terminate.stderr
         and terminate|failed
       changed_when: terminate.stderr == "" and terminate|success
```
Comment (RyanSquared) [inline ldap_ban.yml]: can we `s/|/ is /` to get rid of the Ansible lint warning?

Comment (KellerFuchs) [reply ldap_ban.yml]: Oh, I didn't notice the lint.  How come it didn't fail the CI?

**Comment by KellerFuchs** [comment]
@hashbang/administrators If that's ready to merge, I will rebase and sign, and people who reviewed can sign/merge/...

@RyanSquared, @dpflug Thanks for testing the actual code.
@dpflug Can you still reproduce the failure you had?  I have no idea why it would be failing, complaining about Python strings...

@benharri Thanks for checking the README  <3

**Comment by ghost** [comment]
Smells like my issues are local. https://gist.github.com/dpflug/2e51ea23b71a282d4cc961966cec17a9

**Comment by ghost** [comment]
https://github.com/ansible/ansible/issues/39569 - Found the issue. It's already fixed by PR. I just need to wait for Debian to update Ansible. :skull:

**Comment by ghost** [comment]
Wait, this laptop is Arch. I'm waiting on Ansible 2.7 to release.

That should be much quicker.

**Comment by KellerFuchs** [comment]
@dpflug OK; for a moment I was confused why this doesn't happen here (Ansible 2.6.3 on Debian testing), but it's a Py2 vs. Py3 issue (Py3 support in Ansible still isn't super-good)

**Comment by KellerFuchs** [comment]
@hashbang/administrators That should be ready to go.  If some of the people who reviewed could merge & sign, that would be good.

**Comment by RyanSquared** [comment]
@KellerFuchs PR is still open (sorry!) if you want to amend the docs.

**Comment by KellerFuchs** [comment]
@RyanSquared Indeed, only `README.md` had been updated.
FWIW, `doc/Blocking account` mostly has documentation explaining what the playbook does, but it was still worth updating the “how to run” part.

**Comment by KellerFuchs** [comment]
@RyanSquared Done.

File: ldap_ban.yml (hunk 3 above)
Comment (KellerFuchs) [reply ldap_ban.yml]: I just noticed I missed the same fix in `changed_when`.

**Comment by RyanSquared** [comment]
GitHub isn't showing it as merged, but it is.

**Comment by KellerFuchs** [comment]
@RyanSquared What the heck happened?  Did you rebase before merging?

**Comment by KellerFuchs** [comment]
FYI, the way to do that is rebase the PR branch, (force) push it, then merge to master and push the merge commit.

**Comment by KellerFuchs** [comment]
Merge seems OK:
```
$ git diff --stat ldap_ban 
 roles/coreos-authorized_keys  |  0
 terraform/GNUmakefile         | 21 ---------------------
 terraform/README.md           | 51 +++++++++++++++++++++++----------------------------
 terraform/modules/r53/main.tf |  5 +++--
 4 files changed, 26 insertions(+), 51 deletions(-)
```

**Comment by RyanSquared** [comment]
> @RyanSquared What the heck happened? Did you rebase before merging?

I didn't think I did, but I may have.

**Pull request closed without merging**